plugins/repository-s3/src/main/java/org/opensearch/repositories/s3/S3RepositoryPlugin.java

```diff
@@ -120,9 +135,36 @@ public List<ExecutorBuilder<?>> getExecutorBuilders(Settings settings) {
                 TimeValue.timeValueMinutes(5)
             )
         );
+        executorBuilders.add(
+            new FixedExecutorBuilder(
+                settings,
+                LOW_TRANSFER_QUEUE_CONSUMER,
+                lowPriorityTransferQConsumers(settings),
+                10,
+                "thread_pool." + LOW_TRANSFER_QUEUE_CONSUMER
+            )
+        );
+        executorBuilders.add(
+            new FixedExecutorBuilder(
+                settings,
+                NORMAL_TRANSFER_QUEUE_CONSUMER,
+                normalPriorityTransferQConsumers(settings),
+                10,
+                "thread_pool." + NORMAL_TRANSFER_QUEUE_CONSUMER
+            )
+        );
         return executorBuilders;
     }
 
+    private int lowPriorityTransferQConsumers(Settings settings) {
+        double lowPriorityAllocation = ((double) (100 - S3Repository.S3_PRIORITY_PERMIT_ALLOCATION_PERCENT.get(settings))) / 100;
+        return Math.max(2, (int) (lowPriorityAllocation * S3Repository.S3_TRANSFER_QUEUE_CONSUMERS.get(settings)));
+    }
+
+    private int normalPriorityTransferQConsumers(Settings settings) {
+        return S3Repository.S3_TRANSFER_QUEUE_CONSUMERS.get(settings);
```

Comment:
should normalPriority have S3_PRIORITY_PERMIT_ALLOCATION_PERCENT of total `S3_TRANSFER_QUEUE_CONSUMERS` ?